language/java/mistakes_test.go

```diff
@@ -0,0 +1,121 @@
+package java
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/zimmski/osutil"
+)
+
+func TestMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Logger         *log.Logger
+		RepositoryPath string
+
+		AssertMistakes func(repositoryPath string, mistakes []string)
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			temporaryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryPath, filepath.Base(tc.RepositoryPath))
+			require.NoError(t, osutil.CopyTree(tc.RepositoryPath, repositoryPath))
+
+			buffer, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Log(buffer.String())
+				}
+			}()
+
+			actualMistakes, actualErr := Mistakes(logger, repositoryPath)
+			require.NoError(t, actualErr)
+
+			tc.AssertMistakes(repositoryPath, actualMistakes)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Method without opening brackets",
+
+		RepositoryPath: filepath.Join("..", "..", "testdata", "java", "mistakes"),
+
+		AssertMistakes: func(repositoryPath string, mistakes []string) {
+			assert.Contains(t, mistakes[0], "src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected")
+			assert.Contains(t, mistakes[1], "src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, interface, or enum expected")
```

Comment:
Why contains, and not an exact match?

Reply:
At first I tried the code below which works fine locally:

```go
expectedMistakes := []string{
     repositoryPath + "/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected",
     repositoryPath + "/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, interface, or enum expected",
}

assert.Equal(t, expectedMistakes, mistakes)
```

In the CI:

**Ubuntu**
Works fine

**Macos**
For some reason there is an extra folder in the actual repository path (see logs below)

<details>
 <summary>Log</summary>

```
expected: 
[]string{"/var/folders/lr/439_fwvd3m76p9vy50d57kcc0000gn/T/TestMistakesMethod_without_opening_brackets3653017912/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected", "/var/folders/lr/439_fwvd3m76p9vy50d57kcc0000gn/T/TestMistakesMethod_without_opening_brackets3653017912/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, interface, or enum expected"}
actual : 
[]string{"/private/var/folders/lr/439_fwvd3m76p9vy50d57kcc0000gn/T/TestMistakesMethod_without_opening_brackets3653017912/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected", "/private/var/folders/lr/439_fwvd3m76p9vy50d57kcc0000gn/T/TestMistakesMethod_without_opening_brackets3653017912/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, interface, or enum expected"}
```

</details>

**Windows**
The actual result is not using the windows back slashes struct for file paths but rather the forward slashes, so we would need to convert the repository file path structure for windows tests (see logs below) 

<details>
 <summary>Log</summary>

```
expected: 
[]string{"C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestMistakesMethod_without_opening_brackets3351064100\\001\\mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected", "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestMistakesMethod_without_opening_brackets3351064100\\001\\mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, interface, or enum expected"}
actual : []string{"/C:/Users/RUNNER~1/AppData/Local/Temp/TestMistakesMethod_without_opening_brackets3351064100/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[4,61] ';' expected\r", "/C:/Users/RUNNER~1/AppData/Local/Temp/TestMistakesMethod_without_opening_brackets3351064100/001/mistakes/src/main/java/com/eval/MethodWithoutOpeningBracket.java:[7,1] class, 
```

</details>